doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,621 @@
+
+igraph includes both deterministic and stochastic graph generators (see `generation`). *Deterministic* generators produce the same graph every time you call the fuction, e.g.:
+
+```{r echo = TRUE}
+gi <- make_empty_graph()
+gi <- make_tree(127, 2, mode = "undirected")
+summary(gi)
+```
+
+Uses `make_tree` to generate a regular tree graph with 127 vertices, each vertex having two children (and one parent, of course). No matter how many times you call `make_tree`, the generated graph will always be the same if you use the same parameters:
+
+```{r}
+gi2 <- make_tree(127, 2, mode = "undirected")
+```
+
+```{r echo = TRUE}
+get.edgelist(gi) == get.edgelist(gi2)
+```
+
+The above code snippet also shows you that the `get_edgelist()` method, which returns a list of source and target vertices for all edges, sorted by edge ID. If you print the first 10 elements, you get:
+
+```{r echo = TRUE}
+head(get.edgelist(gi2),10)
+```
+
+*Stochastic* generators produce a different graph each time, e.g. `sample_grg`:
+
+```{r echo = TRUE}
+gi <- make_empty_graph()
+gi <- sample_grg(100, 0.2)
+summary(gi)
+```
+
+This generates a geometric random graph: *n* points are chosen randomly and uniformly inside the unit square and pairs of points closer to each other than a predefined distance *d* are connected by an edge. If you generate GRGs with the same parameters, they will be different:
+
+```{r echo = TRUE}
+gi2 <- make_empty_graph()
+gi2 <- sample_grg(100, 0.2)
+
+gi_a <- head(get.edgelist(gi), 10) 
+gi2_a <-head(get.edgelist(gi2), 10)
+
+gi_a == gi2_a
+```
+
+A slightly looser way to check if the graphs are equivalent is via `isomorphic()`:
+
+```{r echo = TRUE}
+isomorphic(gi,gi2)
+```
+
+Checking for isomorphism can take a while for large graphs (in this case, the answer can quickly be given by checking the degree distributions of the two graphs).
+
+## Setting and retrieving attributes
+
+As mentioned above, in igraph each vertex and each edge have a numeric id from `1` upwards. Deleting vertices or edges can therefore cause reassignments of vertex and/or edge IDs. In addition to IDs, vertex and edges can have *attributes* such as a name, coordinates for plotting, metadata, and weights. The graph itself can have such attributes too (e.g. a name, which will show in `print` or `summary`). In a sense, every `Graph`, vertex and edge can be used as a R variable to store and retrieve these attributes.
+
+To demonstrate the use of attributes, let us create a simple social network:
+
+```{r}
+gi <- make_empty_graph()
+gi <- graph(edges = c(1,2, 1,3, 3,4, 4,5, 5,3, 3,6, 6,1, 7,4, 6,7), directed = FALSE)
+```
+
+Each vertex represents a person, so we want to store names, ages and genders:
+
+```{r echo = TRUE}
+V(gi)$name <- c("Alice", "Bob", "Claire", "Dennis", "Esther", "Frank", "George")
+V(gi)$age <- c(25, 31, 18, 47, 22, 23, 50) 
+V(gi)$gender <- c("f", "m", "f", "m", "f", "m", "m")
+E(gi)$is_formal <- c(FALSE, FALSE, TRUE, TRUE, TRUE, FALSE, TRUE, FALSE, FALSE)
```

Comment:
What is `is_formal` meant to refer to in the context of a social network?

Reply:
This comes straight from the Python tutorial and it is meant to represent whether a connection between two people is a formal or an informal connection.

Reply:
@adriandiazlab will add a short sentence clarifying.